pkg/connector/workspace.go

```diff
@@ -194,6 +194,14 @@ func (o *workspaceResourceType) Grants(ctx context.Context, resource *v2.Resourc
 			rv = append(rv, grant.NewGrant(rr, RoleAssignmentEntitlement, userID))
 		}
 
+		if !user.IsRestricted && user.IsUltraRestricted && !user.IsInvitedUser && !user.IsBot && !user.Deleted {
```

Comment:
is this correct? Missing a `!` on is ultra restricted?

Reply:
I'm almost certain it needs a `!`

```suggestion
		if !user.IsRestricted && !user.IsUltraRestricted && !user.IsInvitedUser && !user.IsBot && !user.Deleted {
```